src/dynimage.rs

```diff
@@ -910,14 +922,21 @@ pub fn load<R: BufRead + Seek>(r: R, format: ImageFormat) -> ImageResult<Dynamic
     }
 }
 
+fn load_guess<R: Read + Seek>(mut r: R) -> ImageResult<DynamicImage> {
+    let mut buffer: [u8; 16] = [0u8; 16];
+    r.read_exact(&mut buffer[..])?;
+    r.seek(SeekFrom::Start(0))?;
```

Comment:
This doesn't seem like the proper behaviour for arbitrary `Seek` impls. The file could be intended to start at a position other than `0`. It should rather use the sequence:

```
let cur = r.seek(SeekFrom::Current(0))?;
r.read_exact(&mut buffer[..])?;
r.seek(SeekFrom::Start(cur))?;
```

For a fixed-length sequence however, it could also be an option to not require `Seek` at all and utilize [`std::io::Chain`](https://doc.rust-lang.org/std/io/struct.Chain.html) on the inspected data.

Reply:
Good point, I didn't consider this. I went ahead with your suggested code.